### PR TITLE
[WIP] Add geocoding limit parameter

### PR DESCRIPTION
**Pull request by KaiBot3000**
Adds the limit parameter for geocoding calls per mapbox/api-geocoder#1102

Tests fail locally, but the same tests are failing on master. I based this on the PR for adding bbox; please let me know if anything is missing.

### CODE REVIEW

**Comment by perrygeo** [comment]
Still TODO

* [ ] add the limit parameter to reverse geocoding

File: docs/datasets.md [hunk 1]
```diff
@@ -45,8 +45,8 @@ You can find it in your account's list of datasets.
 
 ```python
 >>> listing_resp = datasets.list()
->>> new_id in [ds['id'] for ds in listing_resp.json()]
-True
+>>> [ds['id'] for ds in listing_resp.json()]
+[...]
```
Comment (sgillies) [inline docs/datasets.md]: What's the motivation for this change @KaiBot3000 @perrygeo? Tests don't pass otherwise?

It concerns me that we can't assert that a newly added dataset isn't in the list of datasets.

File: docs/geocoding.md [hunk 2]
```diff
@@ -105,9 +105,7 @@ Places at an address may be found using `Geocoder.forward()`.
 'application/vnd.geo+json; charset=utf-8'
 >>> first = response.geojson()['features'][0]
 >>> first['place_name']
-'200 Queen St W, Toronto, Ontario M5T 1T9, Canada'
->>> first['geometry']['coordinates']
-[-79.388951, 43.650579]
+'200 Queen St...'
```
Comment (sgillies) [inline docs/geocoding.md]: 👍

File: docs/geocoding.md [hunk 3]
```diff
@@ -118,7 +116,7 @@ Place results may be biased toward a given longitude and latitude.
 ```python
 
 >>> response = geocoder.forward(
-...     "200 queen street", lon=-66.1, lat=45.3)
+...     "200 queen street", lon=-66.05, lat=45.27)
```
Comment (sgillies) [inline docs/geocoding.md]: Why this change?

Comment (perrygeo) [inline docs/geocoding.md]: The proximity location bias is very sensitive to location precision. Single digit it was returning an address in Australia. Double digit it returned the proper street in Canada. See https://github.com/mapbox/geocoder-feedback/issues/648#issue-182550459

File: tests/test_datasets.py [hunk 4]
```diff
@@ -250,7 +250,7 @@ def test_update_feature():
     """Feature update works."""
 
     def request_callback(request):
-        payload = json.loads(request.body)
+        payload = json.loads(request.body.decode())
```
Comment (sgillies) [inline tests/test_datasets.py]: I don't understand the reason for these changes.

Comment (perrygeo) [inline tests/test_datasets.py]: In python3 `request.body` returns bytes, `json.loads` requires a string.

**Comment by KaiBot3000** [comment]
Hey @sgillies thanks for the review! I wasn't done working on this; it still needs `limit` enabled for reverse geocoding. I'm working on a higher priority quest now but should be able to hit that in the next week. I'll make a new branch for that when it's ready.

**Comment by perrygeo** [comment]
@KaiBot3000 The intent of the PR sorta got buried with all the testing hubub. Sorry about that! I can pick up the remaining work today.

**Comment by KaiBot3000** [comment]
No worries! I'l check back when I'm free and pick up whatever you don't finish :)
